src/db.c

```diff
@@ -227,21 +217,32 @@ robj *lookupKeyWriteOrReply(client *c, robj *key, robj *reply) {
  * The program is aborted if the key already exists. */
 void dbAdd(redisDb *db, robj *key, robj *val) {
     sds copy = sdsdup(key->ptr);
-    dict *d = getDict(db, key->ptr);
+    int slot = getKeySlot(key->ptr);
+    dict *d = db->dict[slot];
+    int was_empty = dictIsEmpty(d);
     dictEntry *de = dictAddRaw(d, copy, NULL);
     serverAssertWithInfo(NULL, key, de != NULL);
     dictSetVal(d, de, val);
     db->key_count++;
+    /* If dict transitioned from empty to non-empty, we should add it to the list of owned slots. */
+    if (was_empty) {
```

Comment:
There is a difference between non-empty and owned.

If we're interested in owned slots, we should instead check e.g. `clusterNodeGetSlotBit(myself, slot)` in cluster.c.

Reply:
I remember there is a use case when we want to ignore owned/not owned during rdb load, as I believe it happens before slots are assigned. I remember I had some test failures related to it when I had explicit slot ownership check in the iterator. I'll need to look into it again and see where we can enforce ownership check and where we'll have to ignore it.

Reply:
maybe the comment goes to to the different terms (just fix the variable name from `owned_slots` to `non_empty_dicts`

Reply:
Done